ledger-core/v2/virtual/object/catalog.go

```diff
@@ -10,13 +10,34 @@ import (
 
 	"github.com/insolar/assured-ledger/ledger-core/v2/conveyor/smachine"
 	"github.com/insolar/assured-ledger/ledger-core/v2/insolar"
+	"github.com/insolar/assured-ledger/ledger-core/v2/log"
+	"github.com/insolar/assured-ledger/ledger-core/v2/reference"
 	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/throw"
 )
 
 type Catalog struct{}
 
-func (p Catalog) Get(_ smachine.ExecutionContext, _ insolar.Reference) SharedStateAccessor {
-	panic(throw.NotImplemented())
+type errEntryMissing struct {
+	*log.Msg `txt:"entry is missing"`
+
+	ObjectReference reference.Global
+}
+
+type errEntryExists struct {
+	*log.Msg `txt:"entry already exists"`
+
+	ObjectReference reference.Global
+}
+
+func formatSMTraceID(ref reference.Global) string {
+	return fmt.Sprintf("object-%s", ref.String())
```

Comment:
the more eficient way to format simple string would be using strings.Builder

Reply:
I don't think it'll be a bottleneck or will affect our performance, but it'll go away completely after we'll incorporate more sophisticated tracing tools